### PR TITLE
Genres/mapper

### PR DESCRIPTION
**Pull request by caroo7**
Entities and repository improvements.
Genres converter and mapper implementation.
Initial spring configuration

### CODE REVIEW

File: database/src/main/java/entities/Genre.java [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    private String name;
+
+    public Genre(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
```
Comment (egolesor) [inline database/src/main/java/entities/Genre.java:25]: You cant make the name final and public ?

File: parser/src/main/java/converter/BookConverter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import mapper.EmpikGenreMapper;
+import mapper.Mapper;
+import repositories.GenreRepository;
+
+import java.sql.Timestamp;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class BookConverter {
```
Comment (egolesor) [inline parser/src/main/java/converter/BookConverter.java:15]: No javadoc. the name BookConventer is very unfamiliar to me :P

File: parser/src/main/java/converter/BookConverter.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    private static final int STANDARD_FIELD_LENGTH = 256;
+
+    private static final int STANDARD_DESCRIPTION_LENGTH = 10000;
+
+    private BookValidator validator = new BookValidator();
+
+    private Mapper genreMapper;
+
+    private GenreRepository genreRepo;
```
Comment (egolesor) [inline parser/src/main/java/converter/BookConverter.java:25]: Why these fields are not final ? are you going to change the reference

File: database/src/main/java/entities/Genre.java (hunk 1 above)
Comment (caroo7) [reply database/src/main/java/entities/Genre.java:25]: Nice remark, probably I can, let's see what hibernate says about that.

File: parser/src/main/java/converter/BookValidator.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package converter;
+
+class BookValidator {
+
+    String validateField(String value, int length) {
```
Comment (egolesor) [inline parser/src/main/java/converter/BookValidator.java:5]: The method validates or prepares the string ?

File: database/src/main/java/entities/Genre.java (hunk 1 above)
Comment (caroo7) [reply database/src/main/java/entities/Genre.java:25]: I have some problems with hibernate after changing to final, so I'll stay with that version.

**Pull request merged**